src/Raven.Server/Documents/Revisions/DeleteRevisionsByDocumentIdMergedCommand.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using Raven.Server.Documents.TransactionMerger.Commands;
+using Raven.Server.ServerWide.Context;
+using Voron;
+
+namespace Raven.Server.Documents.Revisions;
+
+public partial class RevisionsStorage
+{
+    internal sealed class DeleteRevisionsByDocumentIdMergedCommand : AbstractDeleteRevisionsMergedCommand<(bool MoreWork, long Deleted)?>
+    {
+        private readonly List<string> _ids;
+
+        private readonly DateTime? _from, _to;
+
+        public DeleteRevisionsByDocumentIdMergedCommand(List<string> ids, DateTime? from, DateTime? to, bool includeForceCreated) : base(includeForceCreated)
+        {
+            _ids = ids;
+            _from = from;
+            _to = to;
+        }
+
+        protected override (bool MoreWork, long Deleted)? DeleteRevisions(DocumentsOperationContext context)
+        {
+            var revisionsStorage = context.DocumentDatabase.DocumentsStorage.RevisionsStorage;
+            var moreWork = false;
+            var deleted = 0L;
+
+            foreach (var id in _ids)
+            {
+                using (DocumentIdWorker.GetSliceFromId(context, id, out Slice lowerId))
+                using (revisionsStorage.GetKeyPrefix(context, lowerId, out Slice prefixSlice))
+                {
+                    var collectionName = revisionsStorage.GetCollectionFor(context, prefixSlice);
+                    if (collectionName == null)
+                    {
+                        if (revisionsStorage._logger.IsInfoEnabled)
+                            revisionsStorage._logger.Info($"Tried to delete revisions for '{id}' but no revisions found.");
+                        continue;
+                    }
+
+                    var maxDeletes = revisionsStorage.GetRevisionsConfiguration(collectionName.Name, deleteRevisionsWhenNoCofiguration: true)
+                        .MaximumRevisionsToDeleteUponDocumentUpdate;
+
+                    Func<Document, bool> shouldSkip = null;
+                    if (IncludeForceCreated == false || _from.HasValue || _to.HasValue)
+                    {
+                        shouldSkip = ShouldSkipRevision;
+                    }
+
+                    var result = revisionsStorage.ForceDeleteAllRevisionsForInternal(context, lowerId, prefixSlice, collectionName, maxDeletes, shouldSkip);
+                    moreWork |= result.MoreWork;
```

Comment:
not sure this is working as expected.
consider you have here 1k docs with 1k revisions each, would you break after each document? or after the entire 1M revisions will be deleted?

Reply:
You won't stop in either of the two cases you mentioned..
For each document, you'll delete the maximum amount of revisions you can (instructed by the 'MaxDeletesUponUpdate' property of the doc configuration), and if you have documents that 'have more' revisions to delete, you'll execute (by the tx merger) this command again and again until there is no document with remaining revisions anymore.
Same behavior as EnforceRevisionsConfiguration.

Reply:
in the `EnforceRevisionsConfiguration` you will create a new command each time.
here I think you will be looping in the _same_ command blocking the tx merger to do any other job.

Reply:
No, you are creating a new command each time

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-22730/EnfrceConfguration-max-deletes-per-tx

Reply:
Done